pycbc/mchirp_area_improved.py

```diff
@@ -0,0 +1,204 @@
+                                   "and the minimum mass of a black hole. "
+                                   "Used as limits of integration of the "
+                                   "different CBC regions.")
+    mchirp_group.add_argument('--src-class-mchirp-to-delta', type=float,
+                              metavar='m0', default=0.01,
+                              help='Coefficient to estimate the value of the '
+                                   'mchirp uncertainty by mchirp_delta = '
+                                   'm0 * mchirp.')
+    mchirp_group.add_argument('--src-class-eff-to-lum-distance', type=float,
+                              metavar='a0', default=0.759,
+                              help='Coefficient to estimate the value of the '
+                                   'luminosity distance from the minimum '
+                                   'eff distance by D_lum = a0 * min(D_eff).')
+    mchirp_group.add_argument('--src-class-lum-distance-to-delta', type=float,
+                              nargs=2, metavar=('b0', 'b1'),
+                              default=[-0.449, -0.342],
+                              help='Coefficients to estimate the value of the '
+                                   'uncertainty on the luminosity distance '
+                                   'from the estimated luminosity distance and'
+                                   ' the coinc snr by delta_lum = D_lum * '
+                                   'exp(b0) * coinc_snr ** b1.')
+    mchirp_group.add_argument('--src-class-mass-gap-separate',
+                              action='store_true',
+                              help='Gives separate probabilities for each kind'
+                                   ' of mass gap CBC sources: GNS, GG, BHG.')
+
+
+def from_cli(args):
+    return {'mass_limits': {'max_m1': args.src_class_mass_range[1],
+                            'min_m2': args.src_class_mass_range[0]},
+            'mass_bdary': {'ns_max': args.src_class_mass_gap[0],
+                           'gap_max': args.src_class_mass_gap[1]},
+            'estimation_coeff': {'a0': args.src_class_eff_to_lum_distance,
+                                 'b0': args.src_class_lum_distance_to_delta[0],
+                                 'b1': args.src_class_lum_distance_to_delta[1],
+                                 'm0': args.src_class_mchirp_to_delta},
+            'mass_gap': args.src_class_mass_gap_separate}
+
+
+def src_mass_from_z_det_mass(z, del_z, mdet, del_mdet):
+    """Takes values of redshift, redshift uncertainty, detector mass and its
+    uncertainty and computes the source mass and its uncertainty.
+    """
+    msrc = mdet / (1. + z)
+    del_msrc = msrc * ((del_mdet / mdet) ** 2.
+                       + (del_z / (1. + z)) ** 2.) ** 0.5
+    return (msrc, del_msrc)
+
+
+def intmc(mc, x_min, x_max):
+    """Returns the integral of a component mass as a function of the mass of
```

Comment:
maybe the help message could be improved - eg 'returns the integral of m2 over m1 between x_min and x_max, assuming that mchirp is fixed